src/vs/workbench/contrib/debug/browser/debugConfigurationManager.ts

```diff
@@ -715,10 +717,29 @@ class WorkspaceLaunch extends AbstractLaunch implements ILaunch {
 		return this.configurationService.inspect<IGlobalConfig>('launch').workspaceValue;
 	}
 
-	async openConfigFile(sideBySide: boolean, preserveFocus: boolean): Promise<{ editor: IEditorPane | null, created: boolean }> {
+	async openConfigFile(sideBySide: boolean, preserveFocus: boolean, type?: string, token?: CancellationToken): Promise<{ editor: IEditorPane | null, created: boolean }> {
+		const ws = this.contextService.getWorkspace();
+
+		if (ws.folders.length === 0) {
```

Comment:
Is this check necesery?
If I do not have any folders in the workspace I do not even get offered "workspace" in the quick pick

Reply:
That's an edge case I found with the following steps:
1. Open a folder `f1`
2. Add to workspace folder `f2`. An untitled workspace will be created
3. Remove folder `f1` and `f2` from the workspace. At this point we have no folders in the workspace.
4. `this.contextService.getWorkbenchState` returns `WorkbenchState.WORKSPACE` (is this expected?) so the `openConfigFile` function gets called.
Thinking more about this I think I should move that check here https://github.com/microsoft/vscode/blob/7a424f606b2b73547a66e12435a4064dc6b37087/src/vs/workbench/contrib/debug/browser/debugActions.ts#L85

Reply:
Yeah that state is the empty workspace state - you are stil in a workspace just do not have any folder in the workspace. So you are correct we should cover that case. Thanks